Update faq.md (#2189)
prometheus/docs: content/docs/introduction/faq.md

### What is Prometheus?

Prometheus is an open-source systems monitoring and alerting toolkit
with an active ecosystem.
It is the only system directly supported by [Kubernetes](https://kubernetes.io/) and the de facto standard across the [cloud native ecosystem](https://landscape.cncf.io/).
See the [overview](/docs/introduction/overview/).

### How does Prometheus compare against other monitoring systems?

See the [comparison](/docs/introduction/comparison/) page.

### What dependencies does Prometheus have?

> The main Prometheus server runs standalone as a single monolithic binary and has no external dependencies.

Reviewer: This is not correct, we ship other files with prometheus such as the consoles

Reviewer 2: IMO it's correct enough, the console templates are super optional and I count them more as something you can let Prometheus serve than being part of Prometheus or required by it.

Author: I would tend to agree with Julius, which is why I proposed the change. I am also fine not having it, see below.

> The main Prometheus server runs standalone as a single monolithic binary and has no external dependencies.

Author: ```suggestion
The main Prometheus server runs standalone and has no external dependencies.
```

#### Is this cloud native?

Yes.

> Cloud native is a flexible operating model, breaking up old service boundaries to allow for more flexible and scalable deployments.

Reviewer 2: If we want to mention cloud native stuff in here, IMO the main thing making Prometheus cloud native is service discovery integration (to work well in dynamic cloud environments). Of course, also its dimensional data model and ability to store a lot of data.

Author: ```suggestion
Cloud native is a flexible operating model, breaking up old service boundaries to allow for more flexible and scalable deployments.

Prometheus's [service discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) integrates with most tools and clouds. It's dimensional data model and scale into the tens of million of active series allows it to monitor large cloud-native deployments.
```

Author: Like so?

Reviewer 2: Yup! But:

"It's" -> "Its"
"tens of million" -> "tens of millions"

Author: ```suggestion
Cloud native is a flexible operating model, breaking up old service boundaries to allow for more flexible and scalable deployments.

Prometheus's [service discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) integrates with most tools and clouds. Its dimensional data model and scale into the tens of millions of active series allows it to monitor large cloud-native deployments.
```

### I was told Prometheus “doesn't scale”.

This is often more of a marketing claim than anything else.

> A single instance of Prometheus can be more performant than some systems positioning themselves as long term storage solution fo Prometheus.

Reviewer 2: ```suggestion
A single instance of Prometheus can be more performant than some systems positioning themselves as long term storage solution for Prometheus.
```

Pulling over HTTP offers a number of advantages:

* You can start extra monitoring instances as needed, e.g. on your laptop when developing changes.

> * You can more reliably tell if a target is down.

Reviewer 2: I think "easily" was the correct word already - you can be just as reliable with push in telling whether a target is down, but it's harder because now you need to have configurations on both sides about target identities and what to expect.

Author: I was referring to Byzantine faults and near-time detection of failures; probably my ISP roots showing through.

Would "easily and reliably" be a consensus?

Reviewer 2: SGTM

Short answer: Don't! Use something like [Grafana Loki](https://grafana.com/oss/loki/) or [OpenSearch](https://opensearch.org/) instead.

Longer answer: Prometheus is a system to collect and process metrics, not an
event logging system. The Grafana blog post
[Logs and Metrics and Graphs, Oh My!](https://grafana.com/blog/2016/01/05/logs-and-metrics-and-graphs-oh-my/)
provides more details about the differences between logs and metrics.

> If you want to extract Prometheus metrics from application logs, Grafana Loki is designed for just that. See Loki's [metric queries](https://grafana.com/docs/loki/latest/logql/metric_queries/) documentation.

Reviewer: I prefer to recommend mtail. Mtail is lower footprint, as if I "want to extract Prometheus metrics from application logs", that would be my go to solution.

Yes, with the [Alertmanager](https://github.com/prometheus/alertmanager).

> We support sending alerts through email and a webhook system. For currently supported systems, please see our [integrations page](https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver).

Reviewer: There is more natively supported which are not listed in the linked page

Author: ```suggestion
We support sending alerts through email, native integrations, and a webhook system.

Native integrations:
* [OpsGenie](https://www.opsgenie.com/)
* [PagerDuty](http://www.pagerduty.com/)
* [Pushover](https://pushover.net/)
* [Slack](https://slack.com/)
* [VictorOps](https://victorops.com/)
* [WeChat](https://www.wechat.com)
 
For systems supported through webhook, please see our [integrations page](https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver).
```

Reviewer 2: While I commend the effort of listing all Alertmanager notification integrations here, I think that's bound to get out of date... probably we should just link to https://prometheus.io/docs/alerting/latest/configuration/?

Reviewer: it is already out of date

Author: We could say "for direct integrations go here, for webhook go there" and done.

Author: ```suggestion
We support sending alerts through email and a webhook system. For currently supported systems, please see our [integrations page](https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver).

We support sending alerts through [email, various native integrations](https://prometheus.io/docs/alerting/latest/configuration/), and a [webhook system anyone can add integrations to](https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver).
```

Reviewer 2: SGTM, but the first paragraph (line 126 in your snippet above) needs to be removed?

Author: ```suggestion
We support sending alerts through [email, various native integrations](https://prometheus.io/docs/alerting/latest/configuration/), and a [webhook system anyone can add integrations to](https://prometheus.io/docs/operating/integrations/#alertmanager-webhook-receiver).
```
